File: nim/day06.nim
import aoc, strutils

proc solve(path: string): IntSolutions =
  for group in (readFile path).strip.split("\n\n"):
    var any: set[char]
    var all = {'a'..'z'}
    for person in group.splitLines:
      var answers: set[char]
      for c in person: answers.incl c
      any = any + answers
      all = all * answers
    result.first += any.card
    result.second += all.card

echo solve "inputs/06.txt"

File: python/day06.py
from functools import reduce


groups = open('inputs/06.txt').read().split('\n\n')

yes = (lambda group, func:
       len(reduce(func, (set(person) for person in group.split()))))

print(sum(yes(group, set.union) for group in groups))
print(sum(yes(group, set.intersection) for group in groups))
